openstack_controller/datadog_checks/openstack_controller/api/api_rest.py

```diff
@@ -194,6 +194,13 @@ def get_identity_limits(self):
         response.raise_for_status()
         return response.json().get('limits', [])
 
+    def get_block_storage_volumes(self, project_id):
+        response = self.http.get(
+            '{}/volumes/detail'.format(self._catalog.get_endpoint_by_type(Component.Types.BLOCK_STORAGE.value))
+        )
+        response.raise_for_status()
+        return response.json().get('volumes', {})
```

Comment:
Can we add support for pagination? It looks like this endpoint supports it https://docs.openstack.org/api-ref/block-storage/v3/index.html#list-accessible-volumes-with-details

Reply:
Added.